### ISSUE TITLE
Long startup time

### ISSUE THREAD

**Issue by haarp** [jamie-mh/AuthenticatorPro#34]
Hello,

I've been using this app for a while now and I'm super happy with it, especially compared to other authenticator apps. Thanks for making it!

One pain point I encounter tho is long startup times when the app isn't cached. On a Nexus 6 running Android 8 (yes, it's a bit old by now) with version 1.9.0 and 14 entries in the app, I count 5 seconds until the app shows anything, and another 2-3 seconds until the authenticators appear.

Can something be improved here?

Thanks!

**Comment by jamie-mh** [comment]
Hi @haarp ,

Thanks for opening this issue. I've tested the app on older devices and I don't remember experiencing any problems.
Can you answer a few questions that might help me narrow down the issue:

- Do you experience a slow startup on older versions of the app such as 1.8.3?
- Does clearing app data + cache (after creating a backup of course) make a difference?
- Do you use any custom icons?

The app is built with Xamarin so there is an inherant startup lag when compared to standard Android apps but this should only be about 1 second or so.

**Comment by haarp** [comment]
Hey, thanks for your quick reply!

Both 1.8.3 and 1.9.0, even with fresh data/cache, take long to start on ym device. I've never used custom icons.

Cheers

**Comment by jamie-mh** [comment]
Hi, 

I've adjusted a couple of things in the latest release (1.9.1). I can't really tell if it makes a difference as it worked fine on my device before. When you get the update please let me know if there is an improvement.

**Comment by haarp** [comment]
Hello,

thanks for the update. Unfortunately it's still taking about 4 seconds to start and another 3 to show authenticators. Not sure why, tho.

**Comment by Engel1ST** [comment]
Good evening,

I've more 60 OTP in this software with each customs icon and no lag/start up issue on my Huawei P40 PRO.

Alex.

**Comment by jamie-mh** [comment]
> 
> 
> Good evening,
> 
> I've more 60 OTP in this software with each customs icon and no lag/start up issue on my Huawei P40 PRO.
> 
> Alex.

Indeed, I would expect the performance to be good. The p40 pro is a high-end device :)
It seems on some older devices, there is a not insignificant startup delay. I am yet to find the root of the problem though.

**Comment by jamie-mh** [comment]
Following up on this issue, for anyone experiencing this.
In the most recent versions (1.12 and up) has the startup performance been improved over the version mentioned (1.9)?

**Comment by haarp** [comment]
On my old Nexus 6, 1.12.3 is still slow to start up, still around 7 seconds in total. I've since migrated to a Pixel 4, where this app starts up almost instantly.

**Comment by jamie-mh** [comment]
Thanks for getting back to me.
I've recently started testing with an old Moto G (2013 Android 5.1), which wasn't particularly high-end back in its day, and started timing cold starts of the app with 30 authenticators and 2 custom icons.

Here's what I've found:
- With database encryption: 6.29s
- Without database encryption: 3.76s

In the latter case, the performance is somewhat acceptable. If you disable encryption in the app's settings, does this make any difference? Encryption shouldn't provide any benefit in most cases anyway.
